tests/testthat/test_packageToString.r

```diff
@@ -1,125 +1,127 @@
-context("packageToString")
+context("packageToString and stringToPackage")
 
-test_that("packageToString and vice versa", {
-  s = "checkmate"
+# constructs package from string and does basic tests
+test_basic_package = function(s, type, name, string_expect = s, ...) {
   p = stringToPackage(s)
-  expect_is(p, "Package")
-  expect_is(p, "PackageCran")
-  expect_output(print(p), "PackageCran")
-  expect_equal(p$name, "checkmate")
-  expect_equal(packageToString(p), s)
+  expect_is(p, "Package", ...)
+  expect_is(p, type, ...)
+  expect_output(print(p), type, ...)
+  expect_equal(p$name, name, ...)
+  if (!is.na(string_expect)) {
+    p_string = packageToString(p)
+    expect_equal(p_string, string_expect, ...)
+  }
+  invisible(p)
+}
 
-  s = "./assets/package"
-  p = stringToPackage(s)
-  expect_is(p, "Package")
-  expect_is(p, "PackageLocal")
-  expect_output(print(p), "PackageLocal")
-  expect_equal(p$name, "testpkg")
-  expect_equal(p$file_path, normalizePath(s))
-
-  rbuild(s)
-  s = "./assets/testpkg_1.0.tar.gz"
-  on.exit(unlink(s))
-  p = stringToPackage(s)
-  expect_is(p, "Package")
-  expect_is(p, "PackageLocal")
-  expect_output(print(p), "PackageLocal")
-  expect_equal(p$name, "testpkg")
-  expect_equal(p$file_path, normalizePath(s))
-  unlink(s)
+test_that("CRAN Packages", {
+  s = "checkmate"
+test_basic_package(s, "PackageCran", "checkmate")
```

Comment:
if you moved everything to the helpers why not test_basic_package?

In my personal preference `testthat/helper.R` is only for stuff that is potentially used in more then one test R file. So actually this can stay here but I wouldn't have moved the other stuff.

Reply:
I oversee it.
I use a bit different logic: Everything that is a helper function is in `helper.R`. Should we move `test_basic_package` or restore `rinstall`?